File: typescript/src/main/vsdk/toolkit/io/context/ParseSnapshotContext.ts
import { Vector3D } from "../../common/linealAlgebra/Vector3D";
import { Material } from "../../material/Material";
import { Geometry } from "../../skin/Geometry";
import { Patch } from "../../skin/Patch";
import { Vertex } from "../../skin/Vertex";
import { ColorContext } from "./ColorContext";
import { ReaderContext } from "./ReaderContext";
import { TransformStackContext } from "./TransformStackContext";

export class ParseSnapshotContext {
  public currentColor: ColorContext | null;
  public currentFaceList: Patch[] | null;
  public currentGeometryList: Geometry[] | null;
  public currentMaterialName: string | null;
  public currentNormalList: Vector3D[] | null;
  public currentObjectName: string | null;
  public currentPointList: Vector3D[] | null;
  public currentVertexList: Vertex[] | null;
  public currentVertexName: string | null;
  public geometries: Geometry[] | null;
  public geometryStackHeadIndex: number;
  public inComplex: boolean;
  public inSurface: boolean;
  public materials: Material[] | null;
  public monochrome: boolean;
  public singleSided: boolean;
  public warpConeEnds: boolean;
  public numberOfQuarterCircleDivisions: number;
  public readerContext: ReaderContext | null;
  public transformContext: TransformStackContext | null;

  public constructor() {
    this.currentColor = null;
    this.currentFaceList = null;
    this.currentGeometryList = null;
    this.currentMaterialName = null;
    this.currentNormalList = null;
    this.currentObjectName = null;
    this.currentPointList = null;
    this.currentVertexList = null;
    this.currentVertexName = null;
    this.geometries = null;
    this.geometryStackHeadIndex = 0;
    this.inComplex = false;
    this.inSurface = false;
    this.materials = null;
    this.monochrome = false;
    this.singleSided = false;
    this.warpConeEnds = false;
    this.numberOfQuarterCircleDivisions = 0;
    this.readerContext = null;
    this.transformContext = null;
  }
}

File: java/src/main/java/vsdk/toolkit/io/context/ParseSnapshotContext.java
package vsdk.toolkit.io.context;

import java.util.ArrayList;
import vsdk.toolkit.common.linealAlgebra.Vector3D;
import vsdk.toolkit.material.Material;
import vsdk.toolkit.skin.Geometry;
import vsdk.toolkit.environment.geometry.elements.Patch;
import vsdk.toolkit.environment.geometry.elements.Vertex;

public class ParseSnapshotContext {
    // Snapshot of MGF parsing state and outputs (non-owning references).
    public ColorContext currentColor;
    public ArrayList<Patch> currentFaceList;
    public ArrayList<Geometry> currentGeometryList;
    public String currentMaterialName;
    public ArrayList<Vector3D> currentNormalList;
    public String currentObjectName;
    public ArrayList<Vector3D> currentPointList;
    public ArrayList<Vertex> currentVertexList;
    public String currentVertexName;
    public ArrayList<Geometry> geometries;
    public int geometryStackHeadIndex;
    public boolean inComplex;
    public boolean inSurface;
    public ArrayList<Material> materials;
    public boolean monochrome;
    public boolean singleSided;
    public boolean warpConeEnds;
    public int numberOfQuarterCircleDivisions;
    public ReaderContext readerContext;
    public TransformStackContext transformContext;

    public ParseSnapshotContext() {
        currentColor = null;
        currentFaceList = null;
        currentGeometryList = null;
        currentMaterialName = null;
        currentNormalList = null;
        currentObjectName = null;
        currentPointList = null;
        currentVertexList = null;
        currentVertexName = null;
        geometries = null;
        geometryStackHeadIndex = 0;
        inComplex = false;
        inSurface = false;
        materials = null;
        monochrome = false;
        singleSided = false;
        warpConeEnds = false;
        numberOfQuarterCircleDivisions = 0;
        readerContext = null;
        transformContext = null;
    }
}
